### PR TITLE
UCM/API: allow installed package compilation

### PR DESCRIPTION
**Pull request by shssf**
Fix compilation for the OMPI and the user application with the UCM.

### CODE REVIEW

File: src/ucm/api/ucm.h [hunk 1]
```diff
@@ -8,17 +8,17 @@
 #ifndef UCM_H_
 #define UCM_H_
 
-#include <ucs/sys/compiler.h>
+#include <ucs/sys/compiler_def.h>
 
 BEGIN_C_DECLS
```
Comment (shssf) [inline src/ucm/api/ucm.h]: @yosefe Perhaps it is better to remove this line and don't provide `BEGIN_C_DECLS` macro to the user level.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/992/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2957/ for details (Mellanox internal link).

File: src/ucm/api/ucm.h (hunk 1 above)
Comment (yosefe) [inline src/ucm/api/ucm.h]: @shssf we provide this wrapper for all API header files

**Pull request merged**